### PR TITLE
GC: Backup: [FIX]Embedded image requires download

### LINKED ISSUE #2163: Emails with embedded images are not being restored correctly
Emails with embedded images are being restored with broken links.

Repro steps:
1) Create an email with an embedded image (In outlook - send an email to the user with an image pasted)
2) Restore -> Email restores correctly but image has a broken link

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Images copy/pasted into a message were not being saved previously.  Verified with Corso binary

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, in the ChangeLog.

## Type of change

- [x] :bug: Bugfix

## Issue(s)

* closes #2163<issue>

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/connector/support/m365Support.go [hunk 1]
```diff
@@ -71,3 +73,14 @@ func CreateListFromBytes(bytes []byte) (models.Listable, error) {
...
+
+func HasAttachments(body models.ItemBodyable) bool {
+	if body.GetContent() == nil || body.GetContentType() == nil ||
+		*body.GetContentType() == models.TEXT_BODYTYPE || len(*body.GetContent()) == 0 {
+		return false
+	}
+
+	content := *body.GetContent()
+
+	return strings.Contains(content, "src=\"cid:")
```
Comment (gmatev) [inline src/internal/connector/support/m365Support.go:85]: Is this the right check? Should't this check for the image tag primarily = `<img`?

Comment (vkamra) [reply src/internal/connector/support/m365Support.go:85]: The recommendations I've seen are to look for the CID tag and not the 'img' type. I assume there can be other types?

Comment (gmatev) [reply src/internal/connector/support/m365Support.go:85]: Hmmm ok. Wasn't sure if the CID portion is guaranteed to be there.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Check if the user specified for an exchange backup operation has a mailbox.
 - Handle case where user's drive has not been initialized
-
+- Locally stored nline attachments (e.g. copy/paste ) are properly stored ([#2163](https://github.com/alcionai/corso/issues/2163))
```
Comment (vkamra) [inline CHANGELOG.md:14]:
```suggestion
- Inline attachments (e.g. copy/paste ) are discovered and backed up correctly ([#2163](https://github.com/alcionai/corso/issues/2163))
```

File: src/internal/connector/support/m365Support_test.go [hunk 3]
```diff
@@ -159,3 +160,56 @@ func (suite *DataSupportSuite) TestCreateListFromBytes() {
...
+				require.NoError(t, err)
+				return message.GetBody()
+			},
+		},
+		{
+			name:          "Mock w/ inline attachment",
+			hasAttachment: assert.True,
+			getBodyable: func(t *testing.T) models.ItemBodyable {
+				byteArray := mockconnector.GetMessageWithOneDriveAttachment("Test legacy")
+				message, err := CreateMessageFromBytes(byteArray)
```
Comment (vkamra) [inline src/internal/connector/support/m365Support_test.go:189]: @dadams39 - should add a test to `func (suite *GraphConnectorIntegrationSuite) TestRestoreAndBackup()` also please to cover this e2e. The current test only validates the behavior of `HasAttachments`

**Comment by vkamra** [comment]
Adding to mergequeue - should follow up with the e2e test.

**Pull request merged**